File: model/sensor_type.py
import enum

class SensorType(enum.Enum):
    UNKNOWN = (0, 'U', '')
    VISUAL = (1, 'V', 'default')
    INFRARED = (2, 'I', 'Includes Night Vision')
    AIRBORNE = (3, 'A', 'can see over walls cliffs, and other terrain')
    RADAR = (4, 'R', 'Can see through Fog, Rain, Dust, etc')
    ELECTRONIC = (5, 'E', 'detects any sort of electronic emmisions, like radio, cell phones, RADAR, etc')
    
    def __new__(cls, id, abbrev, desc):
        obj = object.__new__(cls)
        obj._value_ = id
        obj.id = id
        obj.abbrev = abbrev
        obj.description = desc
        return obj
    
    @classmethod
    def parse(cls, text):
        text = text.strip().lower()
        if 'eye' == text:
            return cls.VISUAL
        elif 'ir' == text:
            return cls.INFRARED
        elif 'air' == text:
            return cls.AIRBORNE
        elif 'radar' == text:
            return cls.RADAR
        elif 'ew' == text:
            return cls.ELECTRONIC
        else:
            return cls.UNKNOWN

File: src/sensor_type.ts
export class SensorType {

  readonly code: number
  readonly name: string
  readonly description: string

  constructor( c: string, n: string, d: string ){
    this.code = c.toLowerCase().charCodeAt(0)
    this.name = n
    this.description = d
  }

  static readonly UNKNOWN =    new SensorType('?', 'Unknown', '')
  static readonly VISUAL =     new SensorType('V', 'Visual', 'visual-spectrum; default')
  static readonly INFRARED =   new SensorType('I', 'Infrared','Includes Night Vision')
  static readonly AIRBORNE =   new SensorType('A', 'Airborne','can see over walls cliffs, and other terrain')
  static readonly RADAR =      new SensorType('R', 'Radar','Can see through Fog, Rain, Dust, etc')
  static readonly ELECTRONIC = new SensorType('E', 'Electronic','detects any sort of electronic emisions, like radio, cell phones, RADAR, etc')
    
  static parse( text: string ): SensorType {
    text = text.trim().toLowerCase();

    if( 1 == text.length ){
      switch(text.charCodeAt(0)){
        case SensorType.VISUAL.code:
          return SensorType.VISUAL;
        case SensorType.INFRARED.code:
          return SensorType.INFRARED;
        case SensorType.AIRBORNE.code:
           return SensorType.AIRBORNE;
        case SensorType.RADAR.code:
           return SensorType.RADAR;
        case SensorType.ELECTRONIC.code:
          return SensorType.ELECTRONIC;
        default:
          return SensorType.UNKNOWN;
      }
    }else{
      if('eye' === text){
        return SensorType.VISUAL
      }else if('ir' === text){
        return SensorType.INFRARED
      }else if('air' === text){
        return SensorType.AIRBORNE
      }else if('radar' === text){
            return SensorType.RADAR
      }else if('ew' === text){
        return SensorType.ELECTRONIC
      }
      return SensorType.UNKNOWN
    }
  }


};

export default SensorType;
